src/main/java/org/gridsuite/modification/server/NetworkModificationController.java

```diff
@@ -87,13 +91,17 @@ public ResponseEntity<Flux<ModificationInfos>> getModification(@Parameter(descri
         return ResponseEntity.ok().body(networkModificationService.getModification(modificationUuid));
     }
 
-    @PutMapping(value = "/groups/{groupUuid}/modifications/move", produces = MediaType.APPLICATION_JSON_VALUE)
-    @Operation(summary = "Get list of modifications metadata of a group")
-    @ApiResponse(responseCode = "200", description = "List of modifications of the group")
-    public ResponseEntity<Mono<Void>> moveModifications(@Parameter(description = "Group UUID") @PathVariable("groupUuid") UUID groupUuid,
-                                                  @Parameter(description = "before") @RequestParam(value = "before", required = false) UUID before,
-                                                  @Parameter(description = "modification to moves", required = true) @RequestParam(value = "modificationsToMove", required = false) List<UUID> modificationsToMove) {
-        return ResponseEntity.ok().body(networkModificationService.moveModifications(groupUuid, before, modificationsToMove));
+    @PutMapping(value = "/groups/{groupUuid}", produces = MediaType.APPLICATION_JSON_VALUE)
+    @Operation(summary = "For a list of network modifications passed in body, Move them before another one or at the end of the list, or Duplicate them at the end of the list")
+    @ApiResponse(responseCode = "200", description = "The modification list of the group has been updated. Modifications in failure are returned.")
+    public ResponseEntity<Flux<UUID>> updateModificationGroup(@Parameter(description = "group UUID") @PathVariable("groupUuid") UUID groupUuid,
+                                                              @Parameter(description = "kind of modification", required = true) @RequestParam(value = "action") GroupModificationAction action,
+                                                              @Parameter(description = "the modification Uuid to move before (MOVE option, empty means moving at the end)") @RequestParam(value = "before", required = false) UUID before,
+                                                              @RequestBody List<UUID> modificationsUuidList) {
+        if (action == GroupModificationAction.DUPLICATE) {
+            return ResponseEntity.ok().body(networkModificationService.duplicateModifications(groupUuid, modificationsUuidList));
+        }
+        return ResponseEntity.ok().body(networkModificationService.moveModifications(groupUuid, before, modificationsUuidList));
```

Comment:
How about handling unrecognized action value ?

Reply:
Yes, I will do something

Reply:
Actually, it is not possible to call this REST operation with a wrong "action" param, because it is **typed**.
I tried, and the caller (study-server) got an error : `"status":400,"error":"Bad Request","message":"Type mismatch."`